### PR TITLE
Added forward ref.

### PR DESCRIPTION
**Pull request by Ayush2020012016**
Following changes were done:

- [x] Added Forward ref to the headless input 
- [x] Added Forward ref to the ReuseInput 
- [x] Added documentation to the the input.
- [x] Added forward ref to the headless inputgroup
- [x] added forward ref to the reuseinput group
- [x] added the documentation to the input-group

### CODE REVIEW

File: development/input-app/components/input/Forwardrefinput/Forwardrefinput.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const Forwardrefinput = () => {
+    const sampleref = useRef();
+     useEffect(() => {
+      if(sampleref.current){
+        sampleref.current.focus();
+      }
+     }, []);
+    const handleClick = () => {
+        // Access the input value using the ref
+        console.log(">>>",sampleref.current,sampleref.current.value)
```
Comment (rjvim) [inline development/input-app/components/input/Forwardrefinput/Forwardrefinput.jsx:14]: @Ayush2020012016 pl remove this log

File: development/input-app/components/input/Forwardrefinput/Forwardrefinput.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React, { useEffect,Ref, InputHTMLAttributes } from "react";
+import { ReuseInput } from "@locoworks/reusejs-react-input";
+import { useRef } from "react";
+import { HeadlessButton } from "@locoworks/reusejs-react-button";
+const Forwardrefinput = () => {
+    const sampleref = useRef<HTMLInputElement>(null);    
+
+       useEffect(() => {
```
Comment (rjvim) [inline development/input-app/components/input/Forwardrefinput/Forwardrefinput.tsx:8]: @Ayush2020012016 It's not clear if this file is formatted using prettier, indentation seems broken.

File: yarn.lock [hunk 3]
```diff
@@ -1265,116 +1265,6 @@
     human-id "^1.0.2"
```
Comment (rjvim) [inline yarn.lock:1265]: @ShreyDhyani Not sure why so many changes in yarn.lock file, plz check. There are no changes in package.json, but only in yarn.lock file.

**Review by rjvim — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 3 above)
Comment (ShreyDhyani) [reply yarn.lock:1265]: @rjvim These seems to be dependencies for `remark-folder-import` plugin. I checked in Ayush's system and the package `remark-folder-import` is missing, thus its dependencies are not shown. 

He need to take latest pull from main and run yarn again to fix all the missing dependencies issue.

Comment (rjvim) [reply yarn.lock:1265]: @Ayush2020012016

File: components/input/src/HeadlessInput.tsx [hunk 4]
```diff
@@ -1,28 +1,20 @@
...
   showPassword?: boolean;
 }
 
-const HeadlessInput: React.FC<HeadlessInputProps> = ({
-  type,
-  showPassword,
-  ...props
-}) => {
+const HeadlessInput: React.ForwardRefRenderFunction<HTMLInputElement, HeadlessInputProps> = ({ type, showPassword, ...props }, ref) => {
+  console.log(ref)
```
Comment (ShreyDhyani) [inline components/input/src/HeadlessInput.tsx:8]: Remove this console.log

File: components/input/src/ReuseInput.tsx [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
 import HeadlessInput, { HeadlessInputProps } from "./HeadlessInput";
 import { twMerge } from "tailwind-merge";
 
-export interface ReuseInputProps extends HeadlessInputProps {}
+export interface ReuseInputProps extends HeadlessInputProps {
+  inputRef: React.Ref<HTMLInputElement>;
+}
+
+const ReuseInput: React.FC<ReuseInputProps> = ({ className, inputRef, ...rest }) => {
+  console.log({ inputRef });
```
Comment (ShreyDhyani) [inline components/input/src/ReuseInput.tsx:10]: Remove console.logs

File: development/input-app/components/input/Forwardrefinput/Forwardrefinput.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from "react";
```
Comment (ShreyDhyani) [inline development/input-app/components/input/Forwardrefinput/Forwardrefinput.jsx:1]: Rename these files to `ForwardRefInput` Follow the same pattern used in other examples.

File: development/input-app/pages/input/index.mdx [hunk 7]
```diff
@@ -9,6 +9,7 @@ import DisabledInput from "../../components/input/DisabledInput/DisabledInput.ts
 import PasswordInput from "../../components/input/PasswordInput/PasswordInput.tsx";
 import PasswordShowHide from "../../components/input/PasswordShowHide/PasswordShowHide.tsx";
 import NumberInput from "../../components/input/NumberInput/NumberInput.tsx";
+import Forwardrefinput from "../../components/input/Forwardrefinput/Forwardrefinput.jsx"
```
Comment (ShreyDhyani) [inline development/input-app/pages/input/index.mdx:12]: Import and use tsx file here. jsx file is for viewing purpose only.

import Forwardrefinput from "../../components/input/ForwardRefInput/ForwardRefInput.tsx"

File: development/input-group-app/components/input-group/Forwardrefinputgroup/Forwardrefinputgroup.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      }}
+    >
+      Suffix
+    </button>
+  );
+};
+
+const Forwardrefinput = () => {
+    const inputRef = useRef<HTMLInputElement>(null);    
+    console.log(inputRef,"this is the forwardrefinputgruop.jsx")
```
Comment (ShreyDhyani) [inline development/input-group-app/components/input-group/Forwardrefinputgroup/Forwardrefinputgroup.tsx:30]: Remove this console.log

File: development/input-group-app/components/input-group/Forwardrefinputgroup/Forwardrefinputgroup.tsx.preview [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      }}
+    >
+      Suffix
+    </button>
+  );
+};
+
+const Forwardrefinput = () => {
+    const inputRef = useRef();
+    console.log(inputRef,"this is the forwardrefinputgruop.jsx")
```
Comment (ShreyDhyani) [inline development/input-group-app/components/input-group/Forwardrefinputgroup/Forwardrefinputgroup.tsx.preview:30]: Remove log.

**Review by ShreyDhyani — CHANGES_REQUESTED** [review]
@Ayush2020012016 Pls look into the comments mentioned in review.

**Pull request closed without merging**